ouroboros-network/test/Test/Ouroboros/Network/Testnet/Simulation/Node.hs

```diff
@@ -0,0 +1,346 @@
+                         , MonadLabelledSTM m
+                         , MonadCatch       m
+                         , MonadMask        m
+                         , MonadTime        m
+                         , MonadTimer       m
+                         , MonadThrow  (STM m)
+                         , Eq (Async m Void)
+                         , forall a. Semigroup a => Semigroup (m a)
+                         )
+                      => BearerInfo
+                      -> MultiNodeScript
+                      -> m Void
+multinodeSim
+  defaultBearerInfo
+  (MultiNodeScript args) = withAsyncAll (map runNode args) $ \nodes -> do
+    (_, x) <- waitAny nodes
+    return x
+  where
+    runNode :: SimArgs -> m Void
+    runNode ( bgaSlotDuration
+            , bgaSeed
+            , quota
+            , mustReplyTimeout
+            , raps
+            , stdGen
+            , dMap
+            , rap
+            , lrp
+            , peerSelectionTargets
+            , dnsTimeout
+            , dnsLookupDelay
+            ) =
+      withSnocket nullTracer defaultBearerInfo Map.empty
+        $ \ntnSnocket _ -> withSnocket nullTracer defaultBearerInfo Map.empty
+        $ \ntcSnocket _ ->
+          let acceptedConnectionsLimit =
+                AcceptedConnectionsLimit maxBound maxBound 0
+              diffusionMode = InitiatorAndResponderDiffusionMode
+              readPRP = return []
+              readULA = return (UseLedgerAfter 0)
+
+              acceptVersion = \_ v -> Accept v
+
+              defaultMiniProtocolsLimit :: MiniProtocolLimits
+              defaultMiniProtocolsLimit =
+                MiniProtocolLimits { maximumIngressQueue = 64000 }
+
+              blockGeneratorArgs :: BlockGeneratorArgs Block StdGen
+              blockGeneratorArgs =
+                randomBlockGenerationArgs bgaSlotDuration
+                                          bgaSeed
+                                          quota
+
+              stdChainSyncTimeout :: ChainSyncTimeout
+              stdChainSyncTimeout = do
+                  ChainSyncTimeout
+                    { canAwaitTimeout  = shortWait
+                    , intersectTimeout = shortWait
+                    , mustReplyTimeout
+                    }
+
+              limitsAndTimeouts :: Node.LimitsAndTimeouts Block
+              limitsAndTimeouts
+                = Node.LimitsAndTimeouts
+                    defaultMiniProtocolsLimit
+                    (byteLimitsChainSync (const 0))
+                    (timeLimitsChainSync stdChainSyncTimeout)
+                    defaultMiniProtocolsLimit
+                    (byteLimitsKeepAlive (const 0))
+                    timeLimitsKeepAlive
+                    defaultMiniProtocolsLimit
+                    (ProtocolSizeLimits (const smallByteLimit) (const 0))
+                    (ProtocolTimeLimits (const (Just 60)))
+                    defaultMiniProtocolsLimit
+                    (ProtocolSizeLimits (const (4 * 1440))
```

Comment:
It's good to define constants like these at the top level.  This allows to add haddocks, and easily reconfigure the nodes (easier to find), even if it is used just once.